client/src/theme.js

```diff
@@ -6,16 +8,69 @@ const theme = {
     },
     padding: {
       horizontal: '24px',
-      vertical: '8px'
+      vertical: '7px' // 8px - 1px for border
+    },
+    disabled: {
+      opacity: 1
     },
-    extend: (props) => `
-      ${
-        !props.primary &&
+    extend: (props) =>
+      !props.role &&
+      `
+        ${
+          !props.primary &&
+          !props.disabled &&
+          `
+          ;
```

Comment:
This is really cool! But it looks like it's generating invalid CSS, not sure if it's related with these semicolons? Note the final rules for this class, associated with the primary buttons:

```
.hwGyGg{
  display:inline-block;
  box-sizing:border-box;
  cursor:pointer;
  outline:none;
  font:inherit;
  -webkit-text-decoration:none;
  text-decoration:none;
  margin:0;
  background:transparent;
  overflow:visible;
  text-transform:none;
  border:1px solid #017FA3;
  border-radius:4px;
  color:#000000;
  padding:7px 24px;
  font-size:16px;
  line-height:24px;
  background:#017FA3;
  color:#FDFDFD;
  border-radius:4px;
  -webkit-transition-property:color,background-color,border-color,box-shadow;
  transition-property:color,background-color,border-color,box-shadow;
  -webkit-transition-duration:0.1s;
  transition-duration:0.1s;
  -webkit-transition-timing-function:ease-in-out;
  transition-timing-function:ease-in-out;

  false:false;
  color:#FDFDFD;
  -webkit-undefined;
  -ms-flex-undefined;
  undefined;
}
```